android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -74,8 +76,14 @@ fun purchaseProduct(
     if (activity != null) {
         val onReceiveStoreProducts: (List<StoreProduct>) -> Unit = { storeProducts ->
             val productToBuy = storeProducts.firstOrNull {
-                // TODO: Verify this works because "subId:basePlanId" (it should but strings are silly)
-                it.id == productIdentifier && it.type.name.equals(type, ignoreCase = true)
+                val isProduct = (it.id == productIdentifier && it.type.name.equals(type, ignoreCase = true))
+                val isBasePlan = (
+                    it.purchasingData.productId == productIdentifier
+                        && it.googleProduct?.basePlanId == googleBasePlanId
+                        && it.type.name.equals(type, ignoreCase = true)
```

Comment:
do we need to check the type again?

Reply:
also, is checking the type just for safety? or can you set up an OTP and a sub with the same product id?

Reply:
Technically... probably not 🤷‍♂️ Non-subs won't have this but also only subs will have

Can probably remove if you think we should!